### ISSUE TITLE
Rule request: use contains() instead of range(of:) != nil

### ISSUE THREAD

**Issue by ZevEisenberg** [realm/SwiftLint#2776]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Similar to `contains_over_first_not_nil` (https://github.com/realm/SwiftLint/pull/1846) but for strings

1. **Why should this rule be added?** It’s more idiomatic to ask whether the collection contains the element. Checking whether the range of the substring is nil is a more Obj-C way of doing it (e.g. with `range.location == NSNotFound`)
2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Triggering:

```swift
if items.range(of: item) != nil {
    // items contains item
}

if items.range(of: item) == nil { // notice the negation of the previous example
    // items does not contain item
}
```

Not triggering:

```swift
if item.contains(item) {
    // items contains item
}

if !items.contains(item) { // notice the negation of the previous example
    // items does not contain item
}
```
3. **Should the rule be configurable, if so what parameters should be configurable?** I don’t think it needs to be configurable.
4. **Should the rule be opt-in or enabled by default? Why?** I think it should be on by default. It’s always clearer, in my opinion, to express your intent in checking containment, rather than grabbing the index and checking if it’s nil. Plus, it saves on the calculation and storage allocation for the range, which isn’t much, but is probably more than a `Bool`.

**Comment by ZevEisenberg** [comment]
Fixed title. Makes more sense now.
